pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -162,22 +165,27 @@ private File getValidDocIdsSnapshotFile() {
   }
 
   @Override
-  public Dictionary getDictionary(String column) {
+  public <I extends IndexReader> I getIndex(String column, IndexType<?, I, ?> type) {
     ColumnIndexContainer container = _indexContainerMap.get(column);
     if (container == null) {
       throw new NullPointerException("Invalid column: " + column);
     }
-    return container.getDictionary();
+    return type.getIndexReader(container);
+  }
+
+  @Override
+  public Dictionary getDictionary(String column) {
```

Comment:
Do we need these specific methods now that we have a generic `<I extends IndexReader> I getIndex(String column, IndexType<?, I, ?> type)`?